python/cudf/cudf/core/column/column.py

```diff
@@ -306,6 +306,25 @@ def _get_mask_as_column(self) -> ColumnBase:
             self.base_mask, self.offset, self.offset + len(self)
         )
 
+    def _make_copy_replacing_NaT_with_null(self):
```

Comment:
I would still prefer to see this as a free function, outside of a class (not part of `Column`) because it is not a total function of `Column` (it only acts on specific data types) and is only called in the context of another free function `as_column`. I would move this code right above `def as_column` here because that keeps its definition close to the places the function is called: https://github.com/rapidsai/cudf/blob/6ca1f20b632d53c6f31b93ac80cf9fce09228ce0/python/cudf/cudf/core/column/column.py#L1674

To limit the scope even further, we might consider defining this function inside the `as_column` function scope. What do you think?

Reply:
@bdice Following our call, I have implem. `_make_copy_replacing_NaT_with_null()` as a free function in `column.py`, all tests are currently passing without the Type Annotations- being specified so I left it as such. 

Because mypy complained about type-mismatches, otherwise (We can sync on a call to demonstrate)

Reply:
Thanks! Your solution here is good.